source/framework/vx_copy_swap.c

```diff
@@ -0,0 +1,503 @@
+            }
+        }
+    } while ((vx_bool)vx_true_e == node_removed);
+    /* Copy and move node removal - Input nodes.
+     * When ever we remove an Input node, another
+     * node may become an Input node, so we iterate
+     * until no more are found.
+     */
+    do
+    {
+        node_removed = (vx_bool)vx_false_e;
+        /* Look for nodes with input non-virtual and output virtual,
+        and output not connected to a bidirectional of another node */
+        for (i = 0; (i < num_copy_move_nodes) && ((vx_status)VX_SUCCESS == status); i++)
+        {
+            vx_node node = graph->nodes[copy_move_indices[i]];
+            vx_reference first = node->parameters[0];
+            vx_reference second = node->parameters[1];
+            vx_bool removable = second->is_virtual;
+            if ((vx_bool)vx_false_e == first->is_virtual)
+            {
+                /* check for a bidirectional connection in case of a copy node */
+                if ((vx_enum)VX_KERNEL_COPY == node->kernel->enumeration)
+                {
+                    vx_uint32 j;
+                    for (j = 0; j < graph->num_nodes && removable; ++j)
+                    {
+                        if (copy_move_indices[i] != j)
+                        {
+                            vx_node othernode = graph->nodes[j];
+                            vx_uint32 k;
+                            for (k = 0; k < othernode->kernel->signature.num_parameters; ++k)
+                            {
+                                if ((othernode->parameters[k] == second) &&
+                                    ((vx_enum)VX_BIDIRECTIONAL == othernode->kernel->signature.directions[k]))
+                                {
+                                    /* can't remove it */
+                                    removable = (vx_bool)vx_false_e;
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+                if (((vx_bool)vx_true_e == removable) &&
+                    ((vx_bool)vx_true_e == ownCopyMoveReplaceOutputCompatible(first, second)))
+                {
+                    /* we can remove this node, propagating the input reference */
+                    status = ownCopyMoveRemoveNode(graph, copy_move_indices[i], second, first);
+                    if ((vx_status)VX_SUCCESS == status)
+                    {
+                        node_removed = (vx_bool)vx_true_e;
+                        /* and re-make node list list */
+                        num_copy_move_nodes = makeCopyMoveNodeList(graph, copy_move_indices);
+                        /* Process any graph parameter connected to the input */
+                        ownReassignGraphParameter(graph, node, 0, first, (vx_enum)VX_INPUT);
+                        /* re-start the loop */
+                        break;
+                    }
+                    else
```

Comment:
Shall we remove the empty else clause here since this is a simple "if" condition above it and not an "if else"?  Just flagging as a code coverage gap which may be unnecessary.

Reply:
empty else removed